sys/shell/commands/sc_nimble_netif.c

```diff
@@ -241,6 +241,44 @@ static void _cmd_adv(const char *name)
     }
 }
 
+static void _cmd_adv_direct(const char *addr_str)
+{
+    int res;
+    (void)res;
+    uint8_t addrn[BLE_ADDR_LEN];
+    ble_addr_t addr;
+    const struct ble_gap_adv_params _adv_params = {
+        .conn_mode = BLE_GAP_CONN_MODE_DIR,
+        .disc_mode = BLE_GAP_DISC_MODE_GEN,
+        .itvl_min = BLE_GAP_ADV_FAST_INTERVAL2_MIN,
+        .itvl_max = BLE_GAP_ADV_FAST_INTERVAL2_MAX,
+    };
+
+    /* make sure no advertising is in progress */
+    if (nimble_netif_conn_is_adv()) {
+        puts("err: advertising already in progress");
+        return;
+    }
+
+    /* parse and convert address -> RIOT uses big endian notation, NimBLE
+     * expects little endian... */
+    if (bluetil_addr_from_str(addrn, addr_str) == NULL) {
+        puts("err: unable to parse BLE address");
+        return;
+    }
+    addr.type = nimble_riot_own_addr_type;
+    bluetil_addr_swapped_cp(addrn, addr.val);
+
+    /* start advertising directed advertising with the given BLE address */
+    res = nimble_netif_accept_direct(&addr, BLE_HS_FOREVER, &_adv_params);
+    if (res != NIMBLE_NETIF_OK) {
+        printf("err: unable to start directed advertising (%i)\n", res);
+    }
+    else {
+        puts("success: started to send directed advertisements");
+    }
```

Comment:
yes and now. The user can always stop advertising with `ble adv stop`. But for testing it might make sense.

I will look into this together with the timeout for `nimble_netif_accept()`, as a clean integration would probably also require an additional event type...

Reply:
Ok, so I understand this as an out of scope for this one.